### ISSUE TITLE
[Feature request] Ability to configure a retry policy for the client

### ISSUE THREAD

**Issue by ig-sinicyn** [minio/minio-dotnet#387]
Good day here!

We're using the Minio client in our project and everything was fine until we needed to add retries for common failures such as connectivity issues.

Unfortunately, nor the Minio itself nor the RestSharp used by the client do not support ability to setup a retry policy. So, as a temp workaround I've forked the code and added `minioClient.WithRetryPolicy()` method that allows to set up retry policy for the `MinioClient.ExecuteTaskAsync()` method.

The fix does not add dependency on specific retry policy framework; instead, it uses 
```cs
    public delegate Task<IRestResponse> RetryPolicyHandlingDelegate(
        Func<Task<IRestResponse>> executeRequestCallback);
```
delegate to plug-in library of your choice.

[The code is here](https://github.com/ig-sinicyn/minio-dotnet/commit/2c5cf2ee32886fced6113763846560ff0a065ace) and if it is ok, I can do a pull request with it.

**Comment by kannappanr** [comment]
cc @poornas

**Comment by ig-sinicyn** [comment]
Up:)

**Comment by poornas** [comment]
@ig-sinicyn , feel free to send a PR, thanks!

**Comment by poornas** [comment]
Fixed by #392
